src/unix/linux-core.c

```diff
@@ -1045,10 +1252,27 @@ static uint64_t uv__read_cgroups_uint64(const char* cgroup, const char* param) {
 
 
 uint64_t uv_get_constrained_memory(void) {
+  uv__cgroups_subsystem_info_t info;
+  /* For v2 only. */
+  uint64_t max, high;
+
+  if (uv__read_cgroups_proc_files(&info, "memory"))
+    return 0;
   /*
    * This might return 0 if there was a problem getting the memory limit from
    * cgroups. This is OK because a return value of 0 signifies that the memory
    * limit is unknown.
    */
-  return uv__read_cgroups_uint64("memory", "memory.limit_in_bytes");
+  if (CGROUPS_VERSION_1 == info.cgroups_version)
+    return uv__read_cgroups_uint64(info.path, "memory.limit_in_bytes");
+
+  if (CGROUPS_VERSION_2 == info.cgroups_version) {
+    max = uv__read_cgroups_uint64(info.path, "memory.max");
+    high = uv__read_cgroups_uint64(info.path, "memory.high");
+    if (max == 0 || high == 0)
+      return 0;
+    return max > high ? max : high;
```

Comment:
Why does this return the maximum of the two? Shouldn't it return the minumum?

Reply:
Yes, you're right.